### PR TITLE
[bazel] include rust dependencies using crate_universe

### PR DESCRIPTION
**Pull request by jon-flatley**
This PR uses crate_universe under rules_rust to manage rust crate dependencies rather than using bazel build files generated by cargo_raze. This approach makes it easier to modify rust dependencies and removes a large number of checked in autogenerated bazel build files. Rather than running `bazel run //:cargo_raze` one can now run the first build with `CARGO_BAZEL_REPIN=true` after modifying `third_party/rust/crates/Cargo.toml`. Crates added to the `Cargo.toml` can be referenced as `@crate_index://<crate_name>`. For more information check out the documentation in rules_rust https://bazelbuild.github.io/rules_rust/crate_universe.html.

Leaving this as a draft for now for a few reasons:

- [ ] Gather general feedback and consensus that this is the direction we want to go.
- [ ] Determine how this interacts with air-gapped builds.
- [ ] Fix an issue building opentitanlib where the derive macro for serde-annotate's `Annotate` trait doesn't provide an implementation of `thunk_serialize`.

### CODE REVIEW

**Comment by cfrantz** [comment]
This PR is a first iteration in migrating to `crate_universe`.  It uses the `crates_vendor` rule to create BUILD files committed into `//third_party/rust/crates`. 

My primary motivation for choosing this approach is to avoid dealing with the problem of airgapped builds for now and to make the tree of Rust deps very obvious in the codebase.

One of the original issues mentioned by @jon-flatley was a missing implementation of `thunk_serialize` -- This issue is actually a dependency management issue discussed in https://github.com/bazelbuild/rules_rust/discussions/1826.  For now, the best solution seems to be to unify all rust dependencies in the opentitan repo and allow our non-registry crates to pick up those dependencies from opentitan's `@crate_index`.

File: .bazelrc [hunk 1]
```diff
@@ -95,9 +95,12 @@ build:ubsan --strip=never
 build:ubsan --copt -fno-omit-frame-pointer
 build:ubsan --linkopt -fsanitize=undefined
 
+# Enable the rust nightly toolchain
+build --@rules_rust//rust/toolchain/channel=nightly
+
 # Enable the rust 'clippy' linter.
-build --aspects=@rules_rust//rust:defs.bzl%rust_clippy_aspect
-build --output_groups=+clippy_checks
+#build --aspects=@rules_rust//rust:defs.bzl%rust_clippy_aspect
```
Comment (timothytrippel) [inline .bazelrc:102]: should these just be deleted? or are they related to some TODO?

**Review by timothytrippel — APPROVED** [review]
This may cause an issue with airgapped builds, so I am going to pull down these changes and test them before we merge if that is OK?

**Comment by timothytrippel** [comment]
Test look good. We can merge.

Screenshot 2023-02-15 at 11 26 07 AM

**Pull request merged**